packages/syft/src/syft/service/action/action_object.py

```diff
@@ -851,9 +851,40 @@ def _save_to_blob_storage_(self, data: Any) -> SyftError | None:
                 "skipping writing action object to store, passed data was empty."
             )
 
+        return None
+
+    def _save_to_blob_storage(
+        self, allow_empty: bool = False
+    ) -> SyftError | SyftSuccess | SyftWarning:
+        data = self.syft_action_data
+        if isinstance(data, SyftError):
+            return data
+        if isinstance(data, ActionDataEmpty):
+            return SyftError(
+                message=f"cannot store empty object {self.id} to the blob storage"
+            )
+        try:
+            result = self._save_to_blob_storage_(data)
+            if isinstance(result, SyftError | SyftWarning):
+                return result
+            if not TraceResultRegistry.current_thread_is_tracing():
+                self._clear_cache()
+            return SyftSuccess(
+                message=f"Saved action object {self.id} to the blob store"
+            )
+        except Exception as e:
+            print(
+                f"Failed to save action object {self.id} to the blob store. Error: {e}"
+            )
+
         self.syft_action_data_cache = data
```

Comment:
This step should happen under `_save_to_blob_storage_`. Similarly, these three vars should also be set even if saving to memory.

```
self.syft_action_data_type = type(data)
self._set_reprs(data)
self.syft_has_bool_attr = hasattr(data, "__bool__")
```

All of these can still be set under the _save_to_blob_storage_ method beforehand.